java/com/google/re2j/Inst.java

```diff
@@ -84,12 +98,12 @@ boolean matchRune(int r) {
       }
     }
 
-    // Otherwise binary search.
-    for (int lo = 0, hi = runes.length / 2; lo < hi; ) {
+    // Otherwise binary search on rest of the array
+    for (int lo = 0, hi = (runes.length - RUNES_LINER_SEARCH_SIZE) / 2; lo < hi; ) {
```

Comment:
I'm not sure it's worth the fuss to subtract the portion initially scanned.  It's a logarithmic algorithm and this is not the common case. I'm sure it's fast enough to search the complete array.  Did you find evidence to the contrary?

Reply:
no, will remove it